src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/IRCollection.cs

```diff
@@ -398,12 +393,7 @@ internal class IdentityReferenceEnumerator : IEnumerator<IdentityReference>, IDi
 
         internal IdentityReferenceEnumerator(IdentityReferenceCollection collection)
         {
-            if (collection == null)
-            {
-                throw new ArgumentNullException(nameof(collection));
-            }
-
-            _collection = collection;
+            _collection = collection ?? throw new ArgumentNullException(nameof(collection));
```

Comment:
I'd strongly prefer if this one was undone.  (If someone later decided, for some reason, to change the field assignment to alphabetical, there's now an object state change when the exception is thrown).

Reply:
Reverted.